.travis.yml

```diff
@@ -1,27 +1,27 @@
 language: php
-sudo: false
-
-php:
-  - 5.6
-  - 7.0
-  - hhvm
-
 matrix:
+  include:
+    - php: 5.6
+    - php: 7.0
+    - php: 7.1
+      env: PHPUNIT_FLAGS='--coverage-text'
+    - php: hhvm
   fast_finish: true
   allow_failures:
-    - php: 7.0
     - php: hhvm
 
+env:
+  matrix:
+    - COMPOSER_FLAGS="--prefer-lowest"
+    - COMPOSER_FLAGS=""
+
 cache:
   directories:
     - $HOME/.composer/cache
 
-before_install:
-  - travis_retry composer self-update
-
 before_script:
-  - travis_retry composer install --no-interaction
+  - composer update --no-interaction --prefer-dist $COMPOSER_FLAGS
 
 script:
-  - vendor/bin/phpunit --coverage-text
-  - vendor/bin/behat
+  - vendor/bin/phpunit $PHPUNIT_FLAGS
+  - vendor/bin/behat -vv
```

Comment:
I don't think we should put it in verbose mode, however we could do `-f progress --stop-on-failure` and add a `--stop-on-failure` for PHPUnit as well